### PR TITLE
[8.x] Add rails style column definitions in make migration command options

### PR DESCRIPTION
**Pull request by Abdul-fattah-Tayih**
Rails has a very nice [feature](https://guides.rubyonrails.org/active_record_migrations.html#creating-a-migration) which enables you to define columns in the cli when creating a migration, this cuts down time and makes development more snappy.

Especially when creating a quick update or a POC migration, this change will help make it that much quicker to be up and running.

My goal is to reach this:

```console
user@device:~$ php artisan make:migration create_news_table --c=title:string --c=body:text --c=hashed_id:string:index
```

and the result would be:

```php
Schema::create('news', function (Blueprint $table) {
    $table->id();
    $table->string('title');
    $table->text('body');
    $table->string('hashed_id')->index();
    $table->timestamps();
});
```

This change won't break previous version because it adds new functionality to `MigrationMakeCommand` and `MigrationCreator` classes and in case the column definitions were empty, then nothing will happen.

### CODE REVIEW

File: src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php [hunk 1]
```diff
@@ -132,4 +136,26 @@ protected function getMigrationPath()
...
+    {
+        $columns = [];
+        foreach ($this->input->getOption('columns') ?? [] as $column) {
+            $values = explode(':', $column);
+            if (count($values) < 2) {
+                $this->error("$column is invalid, the format is name:type");
+                abort(0);
+            }
+
+
```
Comment (dammy001) [inline src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php:155]: i think you should remove trailing spaces

**Comment by shadoWalker89** [comment]
I'm not against your PR, but I find writing the columns using the power of auto complete more robust (hard to make a typo using auto complete) and time efficient.

File: src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php (hunk 1 above)
Comment (Abdul-fattah-Tayih) [reply src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php]: I've removed the extra empty line

**Comment by Abdul-fattah-Tayih** [comment]
> I'm not against your PR, but I find writing the columns using the power of auto complete more robust (hard to make a typo using auto complete) and time efficient.

You're right in that regard, autocomplete within your IDE/text editor would be more reliable, but there's a check in the implementation to see if the given method exists or not, so the command will tell you if your input was wrong, and personally I see this feature improving developer experience, for example you can use `make:controller Test` and add the usual crud methods manually or use `make:controller Test -r` to have these methods stubbed out for you, I believe this goes the same way.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**